logs/client.go

```diff
@@ -0,0 +1,199 @@
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/solarwinds/swo-cli/version"
+)
+
+type Client struct {
+	opts       *Options
+	httpClient http.Client
+	output     *os.File
+}
+
+type Log struct {
+	Time     time.Time `json:"time"`
+	Message  string    `json:"message"`
+	Hostname string    `json:"hostname"`
+	Severity string    `json:"severity"`
+	Program  string    `json:"program"`
+}
+
+type PageInfo struct {
+	PrevPage string `json:"prevPage"`
+	NextPage string `json:"nextPage"`
+}
+
+type LogsData struct {
+	Logs     []Log `json:"logs"`
+	PageInfo `json:"pageInfo"`
+}
+
+func NewClient(opts *Options) (*Client, error) {
+	return &Client{
+		httpClient: *http.DefaultClient,
+		opts:       opts,
+		output:     os.Stdout,
+	}, nil
+}
+
+func (c *Client) prepareRequest(ctx context.Context, nextPage string) (*http.Request, error) {
+	var logsEndpoint string
+	var err error
+	params := url.Values{}
+	if nextPage == "" {
+		logsEndpoint, err = url.JoinPath(c.opts.ApiUrl, "v1/logs")
+	} else {
+		u, err := url.Parse(nextPage)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse nextPage field: %w", err)
+		}
+
+		logsEndpoint, err = url.JoinPath(c.opts.ApiUrl, u.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err = url.ParseQuery(u.RawQuery)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	logsUrl, err := url.Parse(logsEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.opts.group != "" {
+		params.Add("group", c.opts.group)
+	}
+	if c.opts.minTime != "" {
+		params.Add("startTime", c.opts.minTime)
+	}
+	if c.opts.maxTime != "" {
+		params.Add("endTime", c.opts.maxTime)
+	}
+
+	var filter string
+	if c.opts.system != "" {
+		filter = fmt.Sprintf(`host:"%s"`, c.opts.system)
+	}
+	if len(c.opts.args) != 0 {
+		if len(filter) == 0 {
+			filter = strings.Join(c.opts.args, " ")
+		} else {
+			filter = filter + " " + strings.Join(c.opts.args, " ")
+		}
+	}
+
+	if filter != "" {
+		params.Add("filter", filter)
+	}
+
+	logsUrl.RawQuery = params.Encode()
+	request, err := http.NewRequestWithContext(ctx, "GET", logsUrl.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.opts.Token))
+	request.Header.Add("Accept", "application/json")
+
+	return request, nil
+}
+
+func (c *Client) printResult(logs []Log) error {
+	if c.opts.json {
+		jsonFormat, err := json.Marshal(logs)
+		if err != nil {
+			return err
+		}
+
+		_, err = fmt.Fprintln(c.output, string(jsonFormat))
+		return err
+	}
+
+	for i := len(logs) - 1; i >= 0; i-- {
+		l := logs[i]
+		fmt.Fprintf(c.output, "%s %s %s %s\n", l.Time.Format("Jan 02 15:04:05"), l.Hostname, l.Program, l.Message)
+	}
+
+	return nil
+}
+
+func (c *Client) Run(ctx context.Context) error {
+	if c.opts.version {
+		fmt.Fprintln(c.output, version.Version)
+		return nil
+	}
+
+	var logsCount int
+	var allLogs []Log
+	var nextPage string
+
+	for logsCount < int(c.opts.count) {
```

Comment:
Now that we're looping through each nextPage, this condition should go away:

```suggestion
	for {
```

Otherwise it just stops after the first page.

Reply:
It stops after the first page because `count` is `100` by default and `pageSize` is also `100` by default. I will leave it like this for now until logs api gets fixed.

Reply:
I removed that part, now it stops when there is no more data